jena-arq/src/main/java/org/apache/jena/sparql/engine/iterator/QueryIteratorBase.java

```diff
@@ -70,7 +70,7 @@ protected QueryIteratorBase(AtomicBoolean cancelSignal) {
     }
 
     private boolean requestingCancel() {
-        return requestingCancel != null && requestingCancel.get() ;
+        return (requestingCancel != null && requestingCancel.get()) || Thread.interrupted() ;
```

Comment:
The additional interrupted check is needed in order to cleanly stop the Fuseki server in `TestSPARQLProtocolTimeout` which extends `AbstractFusekiTest`: When shutting down the Fuseki server, Jetty attempts to stop running the remaining threads via interruption. However, so far this flag was not picked up by the `QueryIterator` machinery, causing jetty to abandon the thread and log the warning from `org.eclipse.jetty.util.thread.QueuedThreadPool`:

```
17:27:10 WARN  QueuedThreadPool :: Couldn't stop Thread[qtp1495161082-24,5,main]
```